src/NuGet.Core/NuGet.ProjectModel/LockFile/LockFileTargetLibrary.cs

```diff
@@ -65,7 +70,9 @@ public bool Equals(LockFileTargetLibrary other)
                 && ContentFiles.OrderedEquals(other.ContentFiles, item => item.Path, StringComparer.OrdinalIgnoreCase)
                 && RuntimeTargets.OrderedEquals(other.RuntimeTargets, item => item.Path, StringComparer.OrdinalIgnoreCase)
                 && Build.OrderedEquals(other.Build, item => item.Path, StringComparer.OrdinalIgnoreCase)
-                && BuildMultiTargeting.OrderedEquals(other.BuildMultiTargeting, item => item.Path, StringComparer.OrdinalIgnoreCase);
+                && BuildMultiTargeting.OrderedEquals(other.BuildMultiTargeting, item => item.Path, StringComparer.OrdinalIgnoreCase)
+                && ToolsAssemblies.OrderedEquals(other.ToolsAssemblies, item => item.Path, StringComparer.OrdinalIgnoreCase)
+                && PackageType.OrderedEquals(other.PackageType, item => item);
```

Comment:
Does this cause problems with round tripping? For example if the in-memory lock file has package types set but the on disk one did not write this, does this still work or does it end up re-writing the file?

Reply:
Great call. 

It does. 
I'll look into adding some tests to catch these scenarios.